src/Lecture2.hs

```diff
@@ -150,14 +170,219 @@ You're free to define any helper functions.
        treasure besides gold (if you already haven't done this).
 -}
 
--- some help in the beginning ;)
+newtype Gold = Gold Int
+newtype XP = XP Int
+newtype Health = Health Int deriving (Eq, Ord)
+newtype Attack = Attack Int deriving (Eq, Ord)
```

Comment:
You can define newtypes as records as well. So you won't need to implement `healthToInt` and `attackToInt` functions manually 🙂 

```suggestion
newtype Health = Health { unHealth :: Int } deriving (Eq, Ord)
newtype Attack = Attack { unAttack :: Int } deriving (Eq, Ord)
```